unitary/alpha/quantum_world.py

```diff
@@ -172,17 +285,24 @@ def force_measurement(self, obj: QuantumObject, result: Union[enum.Enum,
         to be a particular result.  A new qubit set to the initial
         state of the result.
         """
-        count = 0
-        ancilla_name = f"ancilla_{obj.name}_{count}"
-        while ancilla_name in self.object_name_dict:
-            count += 1
-            ancilla_name = f"ancilla_{obj.name}_{count}"
-        new_obj = QuantumObject(ancilla_name, result)
-        self.add_object(new_obj)
-        self.ancilla_names.add(ancilla_name)
+        new_obj = self._add_ancilla(namespace=obj.name, value=result)
+        # Swap the input and ancilla qubits using a remapping dict.
```

Comment:
Is there a test case that exercises this new force measurement code?

Reply:
Added a bunch more tests. During those tests, realized that [SparseSimulator does not support qudits at all](#78) but we can support that via this compilation.